src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -141,6 +141,8 @@ static void CreateCitusTable(Oid relationId, CitusTableType tableType,
 							 DistributedTableParams *distributedTableParams);
 static void CreateHashDistributedTableShards(Oid relationId, int shardCount,
 											 Oid colocatedTableId, bool localTableEmpty);
+static void CreateNullShardKeyDistTableShard(Oid relationId, Oid colocatedTableId,
```

Comment:
The naming of these top-level functions should be closer to the user-facing concept than how they are represented in metadata

Reference tables are also null shard key. We could also have insert-optimized distributed tables where the shard key is null, but there are still multiple shards. So the null shard key aspect is more of an implementation detail, and doesn't define this category of tables. I don't think we would talk about null shard key tables in docs.

Conceptually, I think of these more as "SingleShard" tables . You could have distributed tables with 1 shard of course, but that's not a separate category of tables

(We could also get creative, e.g. "Unary" tables)

Reply:
done